packages/spektate/src/Deployment.test.ts

```diff
@@ -0,0 +1,204 @@
+    storageAccountKey: string,
+    storageTableName: string,
+    partitionKey: string,
+    srcPipeline1: IPipeline,
+    hldPipeline1: IPipeline,
+    manifestPipeline: IPipeline,
+    query1?: azure.TableQuery
+  ): Promise<Deployment[]> => {
+    query = query1;
+    return new Promise(resolve => {
+      resolve(rawDeployments);
+    });
+  }
+);
+
+beforeAll(() => {
+  rawDeployments = JSON.parse(
+    fs.readFileSync(mockDirectory + "deployments.json", "utf-8")
+  );
+  srcPipeline.builds = JSON.parse(
+    fs.readFileSync(mockDirectory + "ci-builds.json", "utf-8")
+  );
+  hldPipeline.releases = JSON.parse(
+    fs.readFileSync(mockDirectory + "cd-releases.json", "utf-8")
+  );
+  clusterPipeline.builds = JSON.parse(
+    fs.readFileSync(mockDirectory + "hld-builds.json", "utf-8")
+  );
+  updatePipelineDates();
+});
+
+describe("Deployment", () => {
+  test("Deployments parsing is working as expected", () => {
+    new Promise(resolve => {
+      Deployment.parseDeploymentsFromDB(
+        rawDeployments,
+        srcPipeline,
+        hldPipeline,
+        clusterPipeline,
+        resolve
+      );
+    }).then(value => {
+      expect(value).toHaveLength(62);
+      const deps = value as Deployment[];
+      let verified = false;
+      deps.forEach(dep => {
+        if (dep.deploymentId === "179c843496bd") {
+          expect(dep.status()).toBe("Complete");
+          expect(dep.endTime().getTime()).toBe(
+            new Date("2019-10-31T18:15:53.767Z").getTime()
+          );
+          expect(dep.duration()).toBe("9.24");
+          verified = true;
+        }
+      });
+      expect(verified).toBeTruthy();
+
+      deps.sort(Deployment.compare);
+      expect(deps).toHaveLength(62);
+      expect(
+        deps[61].endTime().getTime() < deps[0].endTime().getTime()
+      ).toBeTruthy();
+      expect(deps[1].endTime().getTime() < deps[0].endTime().getTime());
+    });
+  });
+});
+
+describe("Deployment", () => {
+  test("Query string is built correctly to pull from storage", () => {
+    Deployment.getDeploymentsBasedOnFilters(
+      "",
+      "",
+      "",
+      "",
+      srcPipeline,
+      hldPipeline,
+      clusterPipeline,
+      "Dev",
+      ""
+    );
+    let queryString = JSON.stringify(query);
+    expect(queryString.includes("env eq 'dev'")).toBeTruthy();
+    Deployment.getDeploymentsBasedOnFilters(
+      "",
+      "",
+      "",
+      "",
+      srcPipeline,
+      hldPipeline,
+      clusterPipeline,
+      "",
+      "tag",
+      "211",
+      "abcdefg",
+      "service",
+      "depId"
+    );
+    queryString = JSON.stringify(query);
+    expect(queryString.includes("imageTag eq 'tag'")).toBeTruthy();
+    expect(queryString.includes("p1 eq '211'")).toBeTruthy();
+    expect(queryString.includes("commitId eq 'abcdefg'")).toBeTruthy();
+    expect(queryString.includes("service eq 'service'")).toBeTruthy();
+    expect(queryString.includes("RowKey eq 'depid'")).toBeTruthy();
```

Comment:
@yradsmikham I was under the impression that this tests for the value to be true, but seems like that's not the case, so I'll update these to `toBe(true)` since `toBeTrue()` is not available in jest